### ISSUE TITLE
defining a prop as `xsd:boolean` loses its prefixed URL

### ISSUE THREAD

**Issue by VladimirAlexiev** [digitalbazaar/jsonld.js#558]
Given this context
```js
{"@context":
 {"owl": "http://www.w3.org/2002/07/owl#",
  "xsd": "http://www.w3.org/2001/XMLSchema#",
  "owl:deprecated": {"@type": "xsd:boolean"}
 }
}
```
This instance data
``js
{
  "@id": "ex:Ontology",
  "http://www.w3.org/2002/07/owl#deprecated": true
}
```
is compacted to
```js
{
  "@context": {
    "owl": "http://www.w3.org/2002/07/owl#",
    "xsd": "http://www.w3.org/2001/XMLSchema#",
    "owl:deprecated": {
      "@type": "xsd:boolean"
    }
  },
  "@id": "ex:Ontology",
  "http://www.w3.org/2002/07/owl#deprecated": true
}
```

As you see, the prefix `owl:` is not used for the property.
If I remove the prop type from the context
```js
  "owl:deprecated": {"@type": "xsd:boolean"}
```
Then the prop is shortened.

Same happens [in playground](https://json-ld.org/playground/#startTab=tab-compacted&json-ld=%7B%22%40id%22%3A%22ex%3AOntology%22%2C%22http%3A%2F%2Fwww.w3.org%2F2002%2F07%2Fowl%23deprecated%22%3Atrue%7D&context=%7B%22%40context%22%3A%7B%22owl%22%3A%22http%3A%2F%2Fwww.w3.org%2F2002%2F07%2Fowl%23%22%2C%22xsd%22%3A%22http%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23%22%2C%22owl%3Adeprecated%22%3A%7B%22%40type%22%3A%22xsd%3Aboolean%22%7D%7D%7D) and https://github.com/digitalbazaar/jsonld-cli/releases/tag/v2.0.0

@msporny @gkellogg is this expected?

**Comment by gkellogg** [comment]
Yes, it is operating as designed. Term selection looks for exact patterns, and would have used `owl:deprecated` if it were `"true"^^xsd:boolean`, but it needs to match that value pattern, and not a native value. JSON-LD only takes native types into consideration for from and to RDF transformation.

**Comment by VladimirAlexiev** [comment]
which means that ALL json native values are useless or harmful, and only strings should be used .
Right?

**Comment by gkellogg** [comment]
> which means that ALL json native values are useless or harmful, and only strings should be used . Right?

Pretty much :) native values generally loose fidelity, and the algorithms intentionally ignore them. JSON-LD is really about interpreting string values.
